prototype for enterprise entitlements flow

zubair-arbi opened the pull request:
ENT-84

This is a **sample code** for the prototype of enterprise entitlements.
Please ignore the print statements or commented code.
@asadiqbal08 @saleem-latif @mattdrayer @kashifch

* add settings variable `ENTERPRISE_API_URL` to locate enterprise service API
* add method `enterprise_api_client` in `core/models.py`
* add enterprise utils file to fetch data for enterprises
* update basket view to display enterprise name and logo

**Related edx-enterprise PR for enterprise entitlements:** https://github.com/edx/edx-enterprise/pull/21

### Not Added (pending)
* validate the coupon record in Ecom
* add flow in case learner hasn't signed consent
* apply enterprise coupon in basket view before rendering in case of valid and eligible enterprise learner

### Basket page Screen shots
<img width="1209" alt="screen shot 2016-12-09 at 2 50 28 pm" src="https://cloud.githubusercontent.com/assets/5072991/21044673/f541d65a-be1e-11e6-8534-9ce0e877b76c.png">

mattdrayer commented:
@brittneyexline FYI on this proto-PR

mattdrayer commented on ecommerce/extensions/basket/views.py:
```diff
@@ -106,6 +107,34 @@ def get_context_data(self, **kwargs):
         site_configuration = site.siteconfiguration
+        user = self.request.user
+
+        # TODO: Add Enterprise related entitlements code here
```
We should wrap all of the enterprise code up into its own function(s)

mattdrayer commented on ecommerce/templates/oscar/basket/partials/basket_content.html:
```diff
@@ -11,6 +11,20 @@
+                </span>
+                <span class="pull-left">
+                     {% blocktrans with enterprise_name=enterprise_customer.name %}
+                        You are eligible for the enterprise entitlements from {{ enterprise_name }}.
```
@srpearce -- this is just a prototype at the moment, but just wanted to point out a piece of learner-facing copy here.

mattdrayer commented on ecommerce/templates/oscar/basket/partials/basket_content.html:
```diff
@@ -11,6 +11,20 @@
 
 {% if not basket.is_empty %}
     <div id="content-inner">
+        {% if enterprise_customer %}
```
We should be more specific here about the intended outcome and its corresponding switch -- that is, we probably should not trigger UI behavior based simply on the presence of an object.  If we want to display a logo we should pass a logo, or at least check for the presence of a logo.  If we want to display a line of eligibility copy we should base this on whether or not the user is actually eligible for the entitlement(s).

mattdrayer commented on ecommerce/extensions/basket/views.py:
```diff
@@ -106,6 +107,34 @@ def get_context_data(self, **kwargs):
+            # learner is registered with an enterprise
+            enable_data_sharing_consent = enterprise_customer.get('enable_data_sharing_consent')
+            user_data_sharing_consent = enterprise_customer.get('user_data_sharing_consent')
+            if enable_data_sharing_consent and not user_data_sharing_consent:
```
this might read better if the parameter was named `enterprise_data_sharing_consent_is_required`, and the subsequent parameter `user_data_sharing_consent_is_granted`.

mattdrayer commented on ecommerce/coupons/views.py:
```diff
@@ -100,6 +101,9 @@ def get_context_data(self, **kwargs):
             except exceptions.ProductNotFoundError:
                 return {'error': _('The voucher is not applicable to your current basket.')}
+
+            # TODO: Add Enterprise related entitlements code here
```
Let's be sure to package up this (and all) enterprise-related code into its own function(s) which can be independently covered with tests.

mattdrayer commented on ecommerce/coupons/enterprise_utils.py:
```diff
@@ -0,0 +1,46 @@
+log = logging.getLogger(__name__)
+
+
+def get_user_enterprise_info(user, site):
```
Let's reword all of our enterprise functions to 'get_enterprise...', etc.  For example this one would be `get_enterprise_user_info(user, site)`  I also think that we should position `site` as the first parameter when it is needed.

zubair-arbi commented:
@mattdrayer I have addressed your feedback in a new commit against the @saleem-latif [Ecom PR for entitlements](https://github.com/edx/ecommerce/pull/1046/) in an attempt to centralize the prototype changes:

Please review the changes: https://github.com/edx/ecommerce/pull/1046/commits/450b8f42d626cbfb9a231024192ea4fb1ac7a44e

mattdrayer commented:
Supports work being done in #1061 -- perhaps we should combine these two change sets.

zubair-arbi commented:
Closing this PR as its no longer needed.
